### ISSUE TITLE
It would be good to be able to specify network socket options

### ISSUE THREAD

**Issue by zorn96** [cannatag/ldap3#1016]
We made some changes in 2.5/2.6 to improve the ability of ldap3 to be used in different network environments by allowing things like local bind addresses and local port ranges to use for the local socket to be specified.

it would be good to add a generic option for passing in socket flags/options. socket options are usually 3-tuples of (socket type, option, value), so supporting a list of tuples generically to configure the socket would be a nice improvement for environments where things like keep-alives and idle timeouts might need to differ based on network latency/business

**Comment by stphnlyd** [comment]
Yes it will be good to support socket options. As I mentioned at https://github.com/cannatag/ldap3/issues/1041 at my workspace we have some special case that causes ldap3's default behavior of binding to port 0 difficult in our environment. I think if you can support custom socket options that could also help our case..

**Comment by zorn96** [comment]
@stphnlyd binding to port 0 in python sockets just means "let the system routing decide what port to use dynamically"
`ldap3` already supports an option for specifying the source port if you have restrictions. but if the OS will handle those restrictions, the default should work
